sigstore/_internal/sct.py

```diff
@@ -70,7 +70,7 @@ def verify_sct(
     digitally_signed = _pack_digitally_signed(sct, cert)
     try:
         ctfe_key.verify(
-            signature=sct.signature.encode(),
+            signature=sct.signature,
```

Comment:
Here's what I get when I run this with this change:

```
Traceback (most recent call last):
  File "/home/di/.pyenv/versions/3.11.0a6/bin/sigstore", line 33, in <module>
    sys.exit(load_entry_point('sigstore', 'console_scripts', 'sigstore')())
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/di/.pyenv/versions/3.11.0a6/lib/python3.11/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/di/.pyenv/versions/3.11.0a6/lib/python3.11/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/home/di/.pyenv/versions/3.11.0a6/lib/python3.11/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/di/.pyenv/versions/3.11.0a6/lib/python3.11/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/di/.pyenv/versions/3.11.0a6/lib/python3.11/contextlib.py", line 155, in __exit__
    self.gen.throw(typ, value, traceback)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/di/.pyenv/versions/3.11.0a6/lib/python3.11/site-packages/click/core.py", line 102, in augment_usage_errors
    yield
    ^^^^^
  File "/home/di/.pyenv/versions/3.11.0a6/lib/python3.11/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/di/git/trailofbits/sigstore-python/sigstore/_cli.py", line 26, in _sign
    sign(
    ^^^^^
  File "/home/di/git/trailofbits/sigstore-python/sigstore/_sign.py", line 73, in sign
    verify_sct(sct, cert, ctfe_key)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/di/git/trailofbits/sigstore-python/sigstore/_internal/sct.py", line 72, in verify_sct
    ctfe_key.verify(
    ^^^^^^^^^^^^^^^^
  File "/home/di/.pyenv/versions/3.11.0a6/lib/python3.11/site-packages/cryptography/hazmat/backends/openssl/ec.py", line 367, in verify
    _ecdsa_sig_verify(self._backend, self, signature, data)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/di/.pyenv/versions/3.11.0a6/lib/python3.11/site-packages/cryptography/hazmat/backends/openssl/ec.py", line 102, in _ecdsa_sig_verify
    res = backend._lib.ECDSA_verify(
          ^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: initializer for ctype 'unsigned char *' must be a bytes or list or tuple, not str
```

Why are you getting bytes for `sct.signature`?

Reply:
Additionally, `sct.signature` is Base64 encoded here.

Reply:
@tetsuo-cpp can confirm, but I think he was doing that to get the raw bytes for the signature. That wasn't 100% correct though, since the `signature` attribute was still Base64 encoded. [b79bc1b](https://github.com/trailofbits/sigstore-python/pull/24/commits/b79bc1bd0cdaa6d22e52b14864d29965eb7ba599) should get us a little closer, but I'm still debugging.

Reply:
I might be missing something very obvious, but the `signature` field in the SCT doesn't appear to be a valid DER-encoded DSS signature.

So far I've base64 decoded the SCT, JSON-decoded that, then grabbed the `signature` and base64-decoded *that*, leaving me with a 74-byte long binary string. When I try to throw that string into `openssl asn1parse` it complains about the header being too long.

Reply:
(ECDSA + SHA256 should be 72 bytes, not 74, but removing the first two bytes also doesn't produce a valid DER encoding)

Reply:
Yeah sorry, I didn't push up that change but the diffs I posted in Slack were base 64 decoding the signature. So I have been testing with that change.